bokeh/util/tests/test_serialization.py

```diff
@@ -19,6 +20,46 @@ def test_id_with_simple_ids():
     assert bus.make_id() == "1002"
     del os.environ["BOKEH_SIMPLE_IDS"]
 
+def test_np_consts():
+    assert bus.NP_EPOCH == np.datetime64(0, 'ms')
+    assert bus.NP_MS_DELTA == np.timedelta64(1, 'ms')
+
+def test_binary_array_types():
+    assert len(bus.BINARY_ARRAY_TYPES) == 8
+    for typ in [np.dtype(np.float32),
+                np.dtype(np.float64),
+                np.dtype(np.uint8),
+                np.dtype(np.int8),
+                np.dtype(np.uint16),
+                np.dtype(np.int16),
+                np.dtype(np.uint32),
+                np.dtype(np.int32)]:
+        assert typ in bus.BINARY_ARRAY_TYPES
+
+def test_datetime_types():
+    # includes pandas types during tests
+    assert len(bus.DATETIME_TYPES) == 8
+
+def test_is_datetime_type():
+    assert bus.is_datetime_type(datetime.datetime(2016, 5, 11))
+    assert bus.is_datetime_type(datetime.timedelta(3000))
+    assert bus.is_datetime_type(datetime.date(2016, 5, 11))
+    assert bus.is_datetime_type(datetime.time(3, 54))
+    assert bus.is_datetime_type(np.datetime64("2011-05-11"))
+    assert bus.is_datetime_type(np.timedelta64(3000, 'ms'))
+    assert bus.is_datetime_type(pd.Timedelta("3000ms"))
+    assert bus.is_datetime_type(bus._pd_timestamp(3000000))
+
+def test_convert_datetime_type():
+    assert bus.convert_datetime_type(datetime.datetime(2016, 5, 11)) == 1462942800000.0
+    assert bus.convert_datetime_type(datetime.timedelta(3000)) == 259200000000.0
+    assert bus.convert_datetime_type(datetime.date(2016, 5, 11)) == 1462942800000.0
+    assert bus.convert_datetime_type(datetime.time(3, 54)) == 14040000.0
+    assert bus.convert_datetime_type(np.datetime64("2016-05-11")) == 1462924800000.0
+    assert bus.convert_datetime_type(np.timedelta64(3000, 'ms')) == 3000.0
+    assert bus.convert_datetime_type(pd.Timedelta("3000ms")) == 3000.0
+    assert bus.convert_datetime_type(bus._pd_timestamp(3000000)) == 3.0
```

Comment:
BTW some of these datetime tests for 2016-05-11 seem to give different results depending on the local TZ setting? AFAIK `np.datetime64` is always UTC, but I gather `datetime.datetime` may not be